front-end/src/pages/InboxPage.tsx

```diff
@@ -0,0 +1,150 @@
+import { AxiosWrapper } from '../helpers/AxiosWrapper';
+import React, { useEffect, useState } from 'react';
+import { Row, Col, Form, Input, Nav, NavItem, NavLink, TabContent, Card, CardTitle, TabPane, Container, Button, NavbarToggler, CardBody } from 'reactstrap';
+import LikesFeed from '../components/LikesFeed';
+import PostList from '../components/PostList';
+import { Like } from '../types/Like';
+import { Post } from '../types/Post';
+import { Follow } from '../types/Follow';
+import AuthorListItem from '../components/AuthorListItem';
+import AppSidebar from '../components/AppSidebar';
+var classNames = require('classnames');
+
+
+/**
+ * Renders the homepage which will display a stream of public posts
+ * @param props 
+ */
+// https://stackoverflow.com/questions/44118060/react-router-dom-with-typescript/44571743
+export default function HomePage(props: any) {
+  const [inboxEntries, setInboxEntries] = useState<Post[] | undefined>(undefined);
+  const [likeEntries, setLikeEntries] = useState<Like[]>([]);
+  const [friendReqEntries, setFriendReqEntries] = useState<Follow[] | undefined>(undefined);
+  const [activeTab, setActiveTab] = useState("postsTab");
+  const toggle = (tab: any) => {
+    if (activeTab !== tab) setActiveTab(tab);
+  }
+  const mailIcon = (<svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" height="25" width="25">
+      <path strokeLinecap="round" strokeLinejoin="round" stroke="black" strokeWidth={2} d="M3 8l7.89 5.26a2 2 0 002.22 0L21 8M5 19h14a2 2 0 002-2V7a2 2 0 00-2-2H5a2 2 0 00-2 2v10a2 2 0 002 2z" /></svg>);
+  const peopleIcon = (<svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" height="25" width="25">
+      <path strokeLinecap="round" strokeLinejoin="round" stroke="black" strokeWidth={2} d="M12 4.354a4 4 0 110 5.292M15 21H3v-1a6 6 0 0112 0v1zm0 0h6v-1a6 6 0 00-9-5.197M13 7a4 4 0 11-8 0 4 4 0 018 0z" /></svg>);
+  const likesIcon = (<svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" height="25" width="25">
+      <path strokeLinecap="round" strokeLinejoin="round" stroke="black" strokeWidth={2} d="M4.318 6.318a4.5 4.5 0 000 6.364L12 20.364l7.682-7.682a4.5 4.5 0 00-6.364-6.364L12 7.636l-1.318-1.318a4.5 4.5 0 00-6.364 0z" /></svg>);
```

Comment:
again would like to move these somewhere else if possible

Reply:
I am also trying to figure this out. We have to pay if we want the full package though. Maybe we can make our own module for the icons.

Reply:
I figured how to put all of these in a separate file. Going to push my changes in a bit so you'll see it

Reply:
I'll take a look - I mentioned in a comment above we can install the `react-icons` module: https://github.com/react-icons/react-icons

and instead of having variables, we could just have something like `<mailIcon />`